src/clusterfuzz/_internal/bot/tasks/cron/libs/issue_management/issue_tracker_utils.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities for managing issue tracker instance."""
+
+from clusterfuzz._internal.bot.tasks.cron.libs import request_cache
+from clusterfuzz._internal.bot.tasks.cron.libs.issue_management import \
+    issue_tracker_policy
+from clusterfuzz._internal.bot.tasks.cron.libs.issue_management import jira
+from clusterfuzz._internal.bot.tasks.cron.libs.issue_management import monorail
+from clusterfuzz._internal.config import local_config
+from clusterfuzz._internal.datastore import data_types
+from clusterfuzz._internal.datastore import ndb_utils
+
+_ISSUE_TRACKER_CACHE_CAPACITY = 8
+_ISSUE_TRACKER_CONSTRUCTORS = {
+    'monorail': monorail.get_issue_tracker,
+    'jira': jira.get_issue_tracker
+}
+
+
+def register_issue_tracker(tracker_type, constructor):
+  """Register an issue tracker implementation."""
+  if tracker_type in _ISSUE_TRACKER_CONSTRUCTORS:
+    raise ValueError(
+        'Tracker type {type} is already registered.'.format(type=tracker_type))
+  _ISSUE_TRACKER_CONSTRUCTORS[tracker_type] = constructor
+
+
+def _get_issue_tracker_project_name(testcase=None):
+  """Return issue tracker project name given a testcase or default."""
+  from clusterfuzz._internal.datastore import data_handler
+  job_type = testcase.job_type if testcase else None
+  return data_handler.get_issue_tracker_name(job_type)
+
+
+@request_cache.wrap(capacity=_ISSUE_TRACKER_CACHE_CAPACITY)
```

Comment:
Use memoize.wrap instead for this: https://github.com/google/clusterfuzz/blob/c2de16f9c9ae209a56156f69431e0bc48e608969/src/clusterfuzz/_internal/metrics/fuzzer_stats.py#L361